notification-publisher/src/main/java/org/dependencytrack/notification/publisher/PublishContext.java

```diff
@@ -129,6 +130,9 @@ public static PublishContext fromRecord(final ConsumerRecord<String, Notificatio
         } else if (notification.getSubject().is(VexConsumedOrProcessedSubject.class)) {
             final VexConsumedOrProcessedSubject subject = notification.getSubject().unpack(VexConsumedOrProcessedSubject.class);
             notificationSubjects.put(SUBJECT_PROJECT, Project.convert(subject.getProject()));
+        } else if (notification.getSubject().is(UserPrincipalSubject.class)) {
+            final UserPrincipalSubject subject = notification.getSubject().unpack(UserPrincipalSubject.class);
+            notificationSubjects.put("User", subject);
```

Comment:
The name should be lower-case for consistency. Consider adding a `SUBJECT_USER` constant for it.

Also `notificationSubjects` is meant to appear in logs, and sadly Proto structs tend to be quite "ugly" when stringified (via `.toString()`), so it would be better to either:

* Create a `User` `record` below, like it's done for `Component`, `Project`, etc., or
* Only pass `subject.getUsername()`